CMakeLists.txt

```diff
@@ -0,0 +1,128 @@
+project(opus LANGUAGES C)
+include(opus_config.cmake)
+include(opus_sources.cmake)
+
+add_library(objopus OBJECT ${opus_sources} ${opus_sources_float})
+add_library(opus SHARED $<TARGET_OBJECTS:objopus>)
+add_library(opus_static STATIC $<TARGET_OBJECTS:objopus>)
+
+if(NOT MSVC)
+  set_target_properties(opus_static PROPERTIES OUTPUT_NAME opus)
+endif(NOT MSVC)
+
+target_include_directories(opus
+                           PUBLIC $<BUILD_INTERFACE:
+                                  ${CMAKE_CURRENT_SOURCE_DIR}/include >)
+target_link_libraries(opus PRIVATE ${CMAKE_REQUIRED_LIBRARIES})
+
+target_include_directories(opus_static
+                           PUBLIC $<BUILD_INTERFACE:
+                                  ${CMAKE_CURRENT_SOURCE_DIR}/include >)
+target_link_libraries(opus_static PRIVATE ${CMAKE_REQUIRED_LIBRARIES})
+set_property(TARGET objopus PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+target_include_directories(objopus
+                           PRIVATE $<BUILD_INTERFACE:
+                                   ${CMAKE_CURRENT_SOURCE_DIR}/include
+                                   ${CMAKE_CURRENT_SOURCE_DIR}/celt
+                                   ${CMAKE_CURRENT_SOURCE_DIR}/silk
+                                   >)
+target_compile_definitions(objopus PRIVATE CUSTOM_MODES OPUS_BUILD DLL_EXPORT)
+
+add_sources_group(objopus silk ${silk_sources})
+add_sources_group(objopus celt ${celt_sources})
+
+if(FIXED_POINT)
+  add_sources_group(objopus silk ${silk_sources_fixed})
+  target_include_directories(
+    objopus
+    PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/silk/fixed>)
+  target_compile_definitions(objopus PRIVATE FIXED_POINT=1 DISABLE_FLOAT_API)
+else(FIXED_POINT)
+  add_sources_group(objopus silk ${silk_sources_float})
+  target_include_directories(
+    objopus
+    PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/silk/float>)
+endif(FIXED_POINT)
+
+if(HAVE_SSE)
+  if(FIXED_POINT)
+    add_sources_group(objopus silk ${silk_sources_fixed_sse4_1})
+  else(FIXED_POINT)
+    add_sources_group(objopus silk ${silk_sources_sse4_1})
+  endif(FIXED_POINT)
+
+  add_sources_group(objopus celt ${celt_sources_sse})
+  add_sources_group(objopus celt ${celt_sources_sse2})
+  add_sources_group(objopus celt ${celt_sources_sse4_1})
+
+  if(OPUS_MAY_HAVE_SSE)
+    # Do runtime check for SIMD
+    target_compile_definitions(objopus
+                               PRIVATE
+                               OPUS_X86_MAY_HAVE_SSE
+                               OPUS_X86_MAY_HAVE_SSE2
+                               OPUS_X86_MAY_HAVE_SSE4_1
+                               OPUS_HAVE_RTCD)
+  elseif(OPUS_PRESUME_SSE)
+    target_compile_definitions(objopus
+                               PRIVATE
+                               OPUS_X86_PRESUME_HAVE_SSE
+                               OPUS_X86_PRESUME_HAVE_SSE2
+                               OPUS_X86_PRESUME_SSE4_1)
+  endif()
+
+  if(NOT MSVC)
+    set(CMAKE_C_FLAGS "-msse4.1")
```

Comment:
It makes sense to use a variant of the following macro, as not all compilers support these flags:

```
# Check compiler support for FLAG and set NAME_FLAG_SUPPORTED if supported.
# Add compiler flag to definitions if supported.
macro(check_and_set_flag NAME FLAG)
    include(CheckCXXCompilerFlag)
    check_cxx_compiler_flag(${FLAG} ${NAME}_FLAG_SUPPORTED)
    if (${NAME}_FLAG_SUPPORTED)
        add_definitions(${FLAG})
    endif()
endmacro()
```

Just replace add_definitions() with flag check and `set(CMAKE_C_FLAGS)` conditionally. Use [check_c_compiler_flag](https://cmake.org/cmake/help/latest/module/CheckCCompilerFlag.html?highlight=checkccompilerflag) for C sources, of course.

Reply:
added check for SSE4 support